File: lumina_beauty_final_version/lumina_beauty_backend/LuminaBeauty-Model/src/main/java/pe/edu/pucp/luminaBeauty/Model/Direccion.java
package pe.edu.pucp.luminaBeauty.Model;

public class Direccion {
    private int id;
    private String direccion;
    private String ciudad;
    private String pais; // Agregado (NOT NULL en SQL)
    private String referencia;
    private String codigoPostal; // Agregado
    private boolean esPrincipal;
    private Cliente cliente; // Agregado (La FK obligatoria)

    public Direccion() {
    }

    // Getters y Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getDireccion() { return direccion; }
    public void setDireccion(String direccion) { this.direccion = direccion; }

    public String getCiudad() { return ciudad; }
    public void setCiudad(String ciudad) { this.ciudad = ciudad; }

    public String getPais() { return pais; }
    public void setPais(String pais) { this.pais = pais; }

    public String getReferencia() { return referencia; }
    public void setReferencia(String referencia) { this.referencia = referencia; }

    public String getCodigoPostal() { return codigoPostal; }
    public void setCodigoPostal(String codigoPostal) { this.codigoPostal = codigoPostal; }

    public boolean isEsPrincipal() { return esPrincipal; } // Cambio de get a is (estándar boolean)
    public void setEsPrincipal(boolean esPrincipal) { this.esPrincipal = esPrincipal; }

    public Cliente getCliente() {return cliente;}
    public void setCliente(Cliente cliente) {this.cliente = cliente;}
}

File: lumina_beauty_final_version/lumina_beauty_frontend/LuminaBeauty/Servicios/Modelo/Direccion.cs
using System.Text.Json.Serialization;

namespace LuminaBeauty.Servicios.Modelo
{
    public class Direccion
    {
        [JsonPropertyName("id_direccion")]
        public int IdDireccion { get; set; }

        [JsonPropertyName("direccion")]
        public string DireccionTexto { get; set; } = string.Empty;

        [JsonPropertyName("ciudad")]
        public string Ciudad { get; set; } = string.Empty;

        [JsonPropertyName("pais")]
        public string Pais { get; set; } = string.Empty;

        [JsonPropertyName("referencia")]
        public string Referencia { get; set; } = string.Empty;

        [JsonPropertyName("codigo_postal")]
        public string CodigoPostal { get; set; } = string.Empty;

        [JsonPropertyName("cliente")]
        public Cliente? Cliente { get; set; }
    }
}